extensions/kafka.md

```diff
@@ -1,91 +1,437 @@
+ as otherwise exception upon deserialization should be expected.
+
 ## Configuration in Spring Boot
 
-Axon will automatically provide certain Kafka related components based on the availability of beans and/or properties.
+This extension can be added as a Spring Boot starter dependency to your project using group id
+ `org.axonframework.extensions.kafka` and artifact id `axon-kafka-spring-boot-starter`.
+When using the auto configuration, the following components will be created for you automatically:
+
+**Generic Components:**
+ * A `DefaultKafkaMessageConverter` using the configured `eventSerializer` (which defaults to `XStreamSerializer`).
+ Uses a `String` for the keys and a `byte[]` for the record's values
+ 
+**Producer Components:**
+ * A `DefaultProducerFactory` using a `String` for the keys and a `byte[]` for the record's values.
+ This creates a `ProducerFactory` in confirmation mode "NONE", as is specified [here](#publishing-events-to-kafka).
+ The `axon.kafka.publisher.confirmation-mode` should be adjusted to change this mode,
+  where the "TRANSACTIONAL" mode requires `axon.kafka.producer.transaction-id-prefix` property should be provided.
+ If the `axon.kafka.producer.transaction-id-prefix` is non-null and non-empty,
+  it is assumed a "TRANSACTIONAL" confirmation mode is desired 
+ * A `KafkaPublisher`
+ * A `KafkaEventPublisher`. Assigns the publisher to a processor name and processing group called
+ `__axon-kafka-event-publishing-group` on a `SubscribingEventProcessor`.
+ If a `TrackingEventProcessor` is desired, the `axon.kafka.producer.event-processor-mode` should be set to `tracking`
 
-To enable a KafkaPublisher, either provide a bean of type `ProducerFactory`, or set `axon.kafka.producer.transaction-id-prefix` in `application.properties` to have auto configuration configure a ProducerFactory with Transactional semantics. In either case, `application.properties` should provide the necessary Kafka Client properties, available under the `axon.kafka` prefix. If none are provided, default settings are used, and `localhost:9092` is used as the bootstrap server.
+**Consumer Components:**
+ * A `DefaultConsumerFactory` using a `String` for the keys and a `byte[]` for the record's values
+ * An `AsyncFetcher`. To adjust the `Fetcher`'s poll timeout, the `axon.kafka.fetcher.poll-timeout` can be set.
+ * A `StreamableKafkaMessageSource` which can be used for `TrackingEventProcessor` instances
 
-To enable a `KafkaMessageSource`, either provide a bean of type `ConsumerFactory`, or provide the `axon.kafka.consumer.group-id` setting in `application.properties`. Also make sure all necessary Kafka Client Configuration properties are available under the `axon.kafka` prefix.
+When using the Spring Boot auto configuration be mindful to provide an `application.properties` file.
+The Kafka extension configuration specifics should be placed under prefix `axon.kafka`.
```

Comment:
IMO giving one example of `application.properties` with some basics would add value.

Reply:
Sure, I'll replace the small properties snippet for an more extensive yaml snippet